aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/BaseHandlerStd.java

```diff
@@ -295,20 +304,54 @@ protected ApiVersionDispatcher<ResourceModel, CallbackContext> getApiVersionDisp
         return apiVersionDispatcher;
     }
 
+    protected void validateRequest(final ResourceHandlerRequest<ResourceModel> request) throws RequestValidationException {
+        validateSourceRegion(request);
+    }
+
+    protected void validateSourceRegion(final ResourceHandlerRequest<ResourceModel> request) throws RequestValidationException {
+        final ResourceModel model = request.getDesiredResourceState();
+        final String sourceRegion = model.getSourceRegion();
+        if (StringUtils.isNotBlank(sourceRegion)) {
+            final Set<String> regionNames = Region.regions().stream().map(Region::toString).collect(Collectors.toSet());
+            if (!regionNames.contains(sourceRegion.toLowerCase(Locale.getDefault()))) {
+                throw new RequestValidationException(UNKNOWN_SOURCE_REGION_ERROR);
+            }
+        }
+    }
+
     protected abstract ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ValidatedRequest<ResourceModel> request,
+            final CallbackContext context,
+            final VersionedProxyClient<RdsClient> rdsProxyClient,
+            final VersionedProxyClient<Ec2Client> ec2ProxyClient,
+            final Logger logger
+    );
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
             final AmazonWebServicesClientProxy proxy,
             final ResourceHandlerRequest<ResourceModel> request,
             final CallbackContext context,
             final VersionedProxyClient<RdsClient> rdsProxyClient,
             final VersionedProxyClient<Ec2Client> ec2ProxyClient,
-            final Logger logger);
+            final Logger logger
+    ) {
+        try {
+            validateRequest(request);
+        } catch (RequestValidationException exception) {
+            return ProgressEvent.defaultFailureHandler(exception, HandlerErrorCode.InvalidRequest);
+        }
+
+        return handleRequest(proxy, new ValidatedRequest<ResourceModel>(request), context, rdsProxyClient, ec2ProxyClient, logger);
```

Comment:
Could you elaborate more what is the benefit of `ValidatedRequest` class? I see no extra logic there and validation logic already happened in 
```
             validateRequest(request);
```

Reply:
It is a pure semantic container class. No specific purpose rather than making it explicit that at this stage the validation has been completed. Overloading the corresponding `handleRequest` makes it explicit. In reality, I had some other less conceptual and more down-to-earth reasons. I needed to ensure that the entrypoint to `handleRequest` is sort of guarded. No matter what method overloading I was sketching out, it was clear that the invoker (most likely a test code) still has a way to unconsciously avoid the from-now-on-mandatory request validation. I.e. I needed a more reliable mechanism in place that would ensure the request validation but won't rely on "please call me instead of anything else" approach. Introducing an overloaded class sounded like a decent solution: by a price of an additional object construction we are isolating 2 stages of the request execution: the "unvalidated" and "validated". This also allowed to me make the change with a minimum amount of modifications to the test code. Hope this makes sense.